cmd/jujud/agent/machine.go

```diff
@@ -1048,7 +1065,7 @@ func (a *MachineAgent) initState(agentConfig agent.Config) (*state.StatePool, er
 
 // startModelWorkers starts the set of workers that run for every model
 // in each controller, both IAAS and CAAS.
-func (a *MachineAgent) startModelWorkers(cfg modelworkermanager.NewModelConfig) (worker.Worker, error) {
+func (a *MachineAgent) startModelWorkers(cfg modelworkermanager.NewModelConfig, modelMetrics ModelMetrics) (worker.Worker, error) {
```

Comment:
This sort of change makes it feel 'bolted on'. I can't say I have an immediate better design but it doesn't quite feel right. Why isn't the metrics just part of Config, for example.